### PR TITLE
feat: Add support for renaming tables in schema translator

### PR DESCRIPTION
**Pull request by vermas2012**
This PR allows the users to rename tables while translating their schema from HBase to Bigtable. This is required to translate any tables with custom namespaces as Cloud Bigtable does not support namespaces. 

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

**Review by igorbernstein2 — APPROVED** [review]
lgtm after nits

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java [hunk 1]
```diff
@@ -350,23 +361,123 @@ public HBaseSchemaTranslator(SchemaTranslationOptions options) throws IOExceptio
...
+   * to {@link SchemaWriter}.
+   */
+  interface SchemaTransformer {
+
+    ClusterSchemaDefinition transform(ClusterSchemaDefinition originalSchema)
+        throws IOException, DeserializationException;
+  }
+
+  /** No-op implementation of @{@link SchemaTransformer}. Returns the original schema definition. */
+  static class NoopSchemaTransformer implements SchemaTransformer {
```
[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java:391]: Its package private for testing. I am going to add @VisibleForTesting annotations everywhere.

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java [hunk 2]
```diff
@@ -350,23 +361,123 @@ public HBaseSchemaTranslator(SchemaTranslationOptions options) throws IOExceptio
...
+  }
+
+  /**
+   * Transforms the @{@link ClusterSchemaDefinition} based on a provided JSON map. It can rename
+   * tables before writing them to {@link SchemaWriter}.
+   *
+   * <p>JSON map should look like { "SourceTable": "DestinationTable",
+   * "sourceTable-2":"DestinationTable-2"}
+   */
+  static class JsonBasedSchemaTransformer implements SchemaTransformer {
```
[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java:406]: same

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java:391]: Also we have kept all the static inner classes in the file as package private with @visibleForTesting

[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java]: made all intennal classes either private or package private with (VisibleForTesting)

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java [hunk 3]
```diff
@@ -112,6 +118,7 @@
     String inputFilePath;
     String outputFilePath;
     String tableNameFilter;
+    String schemaMappingFilePath;
```
[the remark this answers is not in the document]
Comment (vermas2012) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java]: Made it Nullable.

**Pull request merged**